### PR TITLE
added makefile

### PR DESCRIPTION
**Pull request by kafkasl**
Added Makefile for convenience

# Definition of DONE
Your changes will be ready for merging after each of the steps below have been completed:

### Testing
- [ ] Unit tests written
- [ ] Tested locally with Inframock
- [ ] Deployed to staging

To set up easy local testing with inframock, follow the instructions here: https://github.com/biomage-ltd/inframock
To deploy to the staging environment, follow the instructions here: https://github.com/biomage-ltd/biomage-utils

### Documentation updates
Is all relevant documentation updated to reflect the proposed changes in this PR?

- [x] Relevant Github READMEs updated
- [x] Relevant wiki pages created/updated

### Approvers
- [ ] Approved by a member of the core engineering team
- [ ] (UX changes) Approved by vickymorrison (this is her username, tag her if you need approval)

### Just before merging:
- [ ] After the PR is approved, the `unstage` script in here: https://github.com/biomage-ltd/biomage-utils is executed. This script cleans up your deployment to staging

### Optional
- [X] Photo of a cute animal attached to this PR

image

### CODE REVIEW

**Comment by kafkasl** [comment]
@ogibson-biomage @alexvpickering if I understood correctly you are the only ones using linux. Using `make run` should no longer need `export DOCKER_GATEWAY_HOST=`docker network inspect --format='{{range .IPAM.Config}}{{.Gateway}}{{end}}'` to work on Linux but I haven't tested it. Could either of you try to `make build && make run`  and see if it works without that enviroment variable set?

**Review by xverges — CHANGES_REQUESTED** [review]

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+# If unix name is not Darwin assume we are on Linux and add this needed env variable
+# See README.md/Running on Docker issues, for more info
+ifneq ($(uname -s), "Darwin")
+	export DOCKER_GATEWAY_HOST=`docker network inspect --format='{{range .IPAM.Config}}{{.Gateway}}{{end}}'
+endif
+#--------------------------------------------------
+# Targets
+#--------------------------------------------------
+install: 
+	@(cd ./local-runner && npm install)
```
[the remark this answers is not in the document]
Comment (kafkasl) [reply Makefile:18]: I have seen `npm install` in all the readmes. Should I update them all? Or why it's been using `npm install` everywhere.

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+# Settings
+#----------------------------------------
+.DEFAULT_GOAL := help
+#--------------------------------------------------
+# Variables
+#--------------------------------------------------
+# If unix name is not Darwin assume we are on Linux and add this needed env variable
+# See README.md/Running on Docker issues, for more info
+ifneq ($(uname -s), "Darwin")
+	export DOCKER_GATEWAY_HOST=`docker network inspect --format='{{range .IPAM.Config}}{{.Gateway}}{{end}}'
```
[the remark this answers is not in the document]
Comment (kafkasl) [reply Makefile:13]: fixed

**Comment by alexvpickering** [comment]
> @ogibson-biomage @alexvpickering if I understood correctly you are the only ones using linux. Using `make run` should no longer need
> 
> ```
> export DOCKER_GATEWAY_HOST=`docker network inspect --format='{{range .IPAM.Config}}{{.Gateway}}{{end}}'
> ```
> 
> to work on Linux but I haven't tested it. Could either of you try to `make build && make run` and see if it works without that enviroment variable set?

Tested `make build && make run`  locally and it worked with caveats:

1) I still have to change `'host.docker.internal'` on line 250 in `init.r` to `'192.168.0.117'` which is the result of `hostname -I |awk '{print $1}'`
2) 
```
DOCKER_GATEWAY_HOST=`docker network inspect --format='{{range .IPAM.Config}}{{.Gateway}}{{end}}'`
```
 doesn't do anything on my linux box:

image

3) Are missing a backtick at the end of the above env variable declaration?

**Comment by kafkasl** [comment]
@alexvpickering Thanks for testing, some questions
> Tested `make build && make run` locally and it worked with caveats:
> 
> 1. I still have to change `'host.docker.internal'` on line 250 and `development_aws_server` on line 24 in `init.r` to `'192.168.0.117'` which is the result of `hostname -I |awk '{print $1}'`

Line 250 of which file? 

> ```
> DOCKER_GATEWAY_HOST=`docker network inspect --format='{{range .IPAM.Config}}{{.Gateway}}{{end}}'`
> ```
> 
> doesn't do anything on my linux box:
> 
> image
> 
> 1. Are you missing a backtick at the end of the above env variable declaration?

Yes, there was a missing backtick, however the point was that exporting that variable was no longer needed, so it doesn't matter.

**Review by ivababukova — COMMENTED** [review]
have you tested it with a linux user?

**Comment by alexvpickering** [comment]
line 250 of `init.r` in pipeline

**Comment by alexvpickering** [comment]
> 1. I still have to change `'host.docker.internal'` on line 250 and `development_aws_server` on line 24 in `init.r` to `'192.168.0.117'` which is the result of `hostname -I |awk '{print $1}'`

both line 250 and 24 or `init.r` in pipeline

**Comment by alexvpickering** [comment]
Thanks @kafkasl looks good to me! I removed the docker stuff from the README

**Review by xverges — APPROVED** [review]

**Pull request merged**